services/src/main/java/org/keycloak/services/resources/account/LinkedAccountsResource.java

```diff
@@ -94,31 +93,47 @@ public LinkedAccountsResource(KeycloakSession session,
         this.user = user;
         realm = session.getContext().getRealm();
     }
-    
+
     @GET
     @Path("/")
     @Produces(MediaType.APPLICATION_JSON)
-    public Response linkedAccounts() {
+    public Response linkedAccounts(
+            @QueryParam("linked") Boolean linked,
+            @QueryParam("keyword") @DefaultValue("") String keyword,
+            @QueryParam("first") @DefaultValue("0") Integer firstResult,
+            @QueryParam("max") @DefaultValue("2147483647") Integer maxResults
+    ) {
         auth.requireOneOf(AccountRoles.MANAGE_ACCOUNT, AccountRoles.VIEW_PROFILE);
-        SortedSet<LinkedAccountRepresentation> linkedAccounts = getLinkedAccounts(this.session, this.realm, this.user);
-        return Cors.add(request, Response.ok(linkedAccounts)).auth().allowedOrigins(auth.getToken()).build();
+
+        List<FederatedIdentityModel> federatedIdentities = session.users().getFederatedIdentitiesStream(realm, user).collect(Collectors.toList());
+        Set<String> socialIds = findSocialIds();
+
+        final String term = keyword.toLowerCase().trim();
+        List<LinkedAccountRepresentation> accounts = realm.getIdentityProvidersStream().filter(IdentityProviderModel::isEnabled)
+                .filter(idp -> {
+                    if(keyword.isEmpty())
```

Comment:
```suggestion
                    if (term.isEmpty())
```

Term might be better as it is already trimmed. In case keyword is something like "  ", it doesn't need to be filtered IMO